client/tink_keyset_helper.go

```diff
@@ -119,3 +130,97 @@ func readTinkKeysetFromBytes(data []byte) (*tinkpb.Keyset, error) {
+		// the data of each version is a tink keyset that contains a single tink key
+		keysetContainsASingleKey, err := readTinkKeysetFromBytes(v.Data)
+		if err != nil {
+			return nil, nil, err
+		}
+		singleKey := keysetContainsASingleKey.Key[0]
+		if v.Status == knox.Primary {
+			tinkKeyset.PrimaryKeyId = singleKey.KeyId
+		}
+		tinkKeyset.Key = append(tinkKeyset.Key, singleKey)
+		tinkKeyIDToKnoxVersionID[singleKey.KeyId] = v.ID
+	}
+	// Convert tink keyset to tink keyset handle. tink doesn't allow transferring cleartext keyset to keyset
+	// handle driectly. Hence, keyset is converted to bytes, then read by package insecurecleartextkeyset
+	bytesBuffer := new(bytes.Buffer)
+	writer := keyset.NewBinaryWriter(bytesBuffer)
+	writer.Write(tinkKeyset)
+	reader := keyset.NewBinaryReader(bytesBuffer)
+	keysetHandle, err := insecurecleartextkeyset.Read(reader)
+	if err != nil {
+		fatalf("cannot get tink keyset handle: %v", err)
+	}
+	return keysetHandle, tinkKeyIDToKnoxVersionID, nil
+}
+
+// getKeysetInfoFromTinkKeysetHandle returns a string representation of the info of the given tink keyset
+// handle. The returned string which does not contain any sensitive key material.
+func getKeysetInfoFromTinkKeysetHandle(
+	keysetHandle *keyset.Handle,
+	tinkKeyIDToKnoxVersionID map[uint32]uint64,
+) (string, error) {
+	// translate the info from the tink build-in function to json format
+	keysetInfo := newJSONTinkKeysetInfo(keysetHandle.KeysetInfo(), tinkKeyIDToKnoxVersionID)
+	keysetInfoForPrint, err := json.MarshalIndent(keysetInfo, "", "  ")
+	if err != nil {
+		return "", err
+	}
+	return string(keysetInfoForPrint), nil
+}
+
+// JSONTinkKeysetInfo translates tink keyset info to JSON format, doesn't contain any actual key material.
+type JSONTinkKeysetInfo struct {
+	PrimaryKeyId uint32                        `json:"primary_key_id"`
+	KeyInfo      []*JSONTinkKeysetInfo_KeyInfo `json:"key_info"`
+}
+
+// JSONTinkKeysetInfo_KeyInfo translates tink key info to JSON format, doesn't contain any actual key material.
+type JSONTinkKeysetInfo_KeyInfo struct {
+	TypeUrl          string `json:"type_url"`
+	Status           string `json:"status"`
+	KeyId            uint32 `json:"key_id"`
+	OutputPrefixType string `json:"output_prefix_type"`
+	KnoxVersionID    uint64 `json:"knox_version_id"`
+}
+
+// newJSONTinkKeysetInfo translates Tink keyset info to JSON format.
+func newJSONTinkKeysetInfo(
+	keysetInfo *tinkpb.KeysetInfo,
+	tinkKeyIDToKnoxVersionID map[uint32]uint64,
+) JSONTinkKeysetInfo {
+	return JSONTinkKeysetInfo{
+		keysetInfo.PrimaryKeyId,
+		newJSONTinkKeysetInfo_KeyInfo(keysetInfo.KeyInfo, tinkKeyIDToKnoxVersionID),
+	}
+}
+
+// newJSONTinkKeysetInfo_KeyInfo translates Tink key info to JSON format.
+func newJSONTinkKeysetInfo_KeyInfo(
+	keyseInfo_KeyInfo []*tinkpb.KeysetInfo_KeyInfo,
+	tinkKeyIDToKnoxVersionID map[uint32]uint64,
+) []*JSONTinkKeysetInfo_KeyInfo {
+	var tinkKeyInfo []*JSONTinkKeysetInfo_KeyInfo
+	for _, v := range keyseInfo_KeyInfo {
```

Comment:
typo in var name? "keysetInfo_KeyInfo".  

In addition _ in variable names is pretty non-standard in Golang, but that should be addressed by updating the struct names (see above comment)

Reply:
Thank you for your comments! I have revised it. I name it refer to [Tink golang package](https://github.com/google/tink/blob/7c93a224b8fa6a3babfaf71c18c5610052dcbd61/go/proto/tink_go_proto/tink.pb.go#L451).

Reply:
I don't see the change, have you committed it?